test/unit/test_session_manager.py

```diff
@@ -19,7 +28,7 @@
 
 mock_conn = mock.Mock()
 mock_conn.disable_request_pooling = False
-mock_conn._ocsp_mode = lambda: True
+mock_conn._ocsp_mode = lambda: DEFAULT_OCSP_MODE
```

Comment:
This was the source of the hanging, during tear-down we tried to do `._ocsp_mode().name` but `bool` doesn't have a `name` method in it. This was then infinitely retried.
We should ramp up the efforts to introduce `mypy` in our code-base to catch errors like this in the future.